tests/rust/tcp-test/connect/mod.rs

```diff
@@ -168,3 +227,22 @@ fn connect_accepting_socket(libos: &mut LibOS, local: &SocketAddrV4, remote: &So
 
     Ok(())
 }
+
+/// Attempts to connect a TCP socket that is closed.
+fn connect_closed_socket(libos: &mut LibOS, remote: &SocketAddrV4) -> Result<()> {
```

Comment:
Not sure if we should add tests for async_close or just eventually remove close(). Close needs to eventually be an async because it is blocking in general and we don't want to block execution to wait for close to complete.

Reply:
I agree that once we have `async_close` implemented, `close` should be dropped alogside all its tests.